src/Symfony/Component/HttpKernel/Controller/ArgumentResolver/PinnedResolverValueResolver.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpKernel\Controller\ArgumentResolver;
+
+use Psr\Container\ContainerInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpKernel\Attribute\ValueResolver;
+use Symfony\Component\HttpKernel\Controller\ArgumentValueResolverInterface;
+use Symfony\Component\HttpKernel\Controller\ValueResolverInterface;
+use Symfony\Component\HttpKernel\ControllerMetadata\ArgumentMetadata;
+use Symfony\Component\HttpKernel\Exception\ResolverNotFoundException;
+use Symfony\Contracts\Service\ServiceProviderInterface;
+
+/**
+ * Delegates value resolving of arguments bearing the {@see ValueResolver} attribute to the specified resolver.
+ *
+ * @author Mathieu Lechat <mathieu.lechat@les-tilleuls.com>
+ */
+final class PinnedResolverValueResolver implements ValueResolverInterface
+{
+    public function __construct(private readonly ContainerInterface $resolvers)
+    {
+    }
+
+    public function resolve(Request $request, ArgumentMetadata $argument): iterable
+    {
+        if (!$attributes = $argument->getAttributesOfType(ValueResolver::class, ArgumentMetadata::IS_INSTANCEOF)) {
+            return [];
+        }
+
+        $resolverName = null;
+        foreach ($attributes as $attribute) {
+            if (!$attribute->pinned) {
+                continue;
+            }
+
+            if ($resolverName) {
+                throw new \LogicException(sprintf('You can only pin one resolver per argument, but $%s has more.', $argument->getName()));
+            }
+
+            $resolverName = $attribute->name;
+        }
+
+        if (!$resolverName) {
+            return [];
+        }
+
+        if (!$this->resolvers->has($resolverName)) {
+            throw new ResolverNotFoundException($resolverName, $this->resolvers instanceof ServiceProviderInterface ? array_keys($this->resolvers->getProvidedServices()) : []);
+        }
+        $resolver = $this->resolvers->get($resolverName);
+        if (!$resolver instanceof ArgumentValueResolverInterface && !$resolver instanceof ValueResolverInterface) {
+            throw new \DomainException(sprintf('"%s" resolver must implement "%s" or "%s", but got "%s".', $resolverName, ArgumentValueResolverInterface::class, ValueResolverInterface::class, get_debug_type($resolver)));
+        }
+
+        $resolvedValue = $resolver->resolve($request, $argument);
+        if ($resolvedValue instanceof \Traversable) {
+            $resolvedValue = iterator_to_array($resolvedValue);
+        }
+        if (!$resolvedValue) {
+            throw new \UnexpectedValueException(sprintf('"%s" resolver did not yield any value for the "%s" argument.', $resolverName, $argument->getName()));
```

Comment:
as mentioned before, I think this check is blocking legit use cases, it would be both simpler and more powerful to return on L65 directly

Reply:
Unless I’m missing something, pinning a resolver to have your argument resolved by another feels neither simple nor powerful to me.

Increasing the priority of the resolver you want would have the same effect with the advantage it would be run once instead of twice.